File: src_simple/f90/factorielle.f90
! factorielle  -  11.03.2014

! ******************************************
! Calcul de factorielle n
! ******************************************

program factorielle

    implicit none

    integer :: i, n, facto

    ! lecture de n
    write(*, *) "Entrer n :"
    read(*, *) n
    write(*, *) "n = ", n

    ! initialiation
    facto = 1

    ! calcul de la factorielle
    do i = 2, n
        facto = facto * i
    end do

    ! affichage résultat
    write(*, *) "Résultat = ", facto

end program factorielle

File: src_simple/python/factorielle.py
#!/usr/bin/env python
# -*- coding=utf-8 -*-

""" factorielle  -  11.03.2014 """

def factorielle(n):
    """ Calcul de factorielle n """
    
    if n == 0:
        factorielle = 1
    else:
        # initialisation
        factorielle = 1

        # calcul
        for i in range(2, n + 1):
            factorielle *= i

    return factorielle

if __name__ == "__main__":
    # lecture de n
    n = int(raw_input("entrer n : "))
    print("n = {0}".format(n))

    # affichage du résultat
    print("{0}! = {1}".format(n, factorielle(n)))

